src/py/flwr/server/driver/driver.py

```diff
@@ -69,44 +70,185 @@ def _get_grpc_driver_and_run_id(self) -> Tuple[GrpcDriver, int]:
+        self,
+        content: RecordSet,
+        message_type: str,
+        dst_node_id: int,
+        group_id: str,
+        ttl: str,
+    ) -> Message:
+        """Create a new message with specified parameters.
+
+        This method constructs a new `Message` with given content and metadata.
+        The `run_id` and `src_node_id` will be set automatically.
+
+        Parameters
+        ----------
+        content : RecordSet
+            The content for the new message. This holds records that are to be sent
+            to the destination node.
+        message_type : str
+            The type of the message, defining the action to be executed on
+            the receiving end.
+        dst_node_id : int
+            The ID of the destination node to which the message is being sent.
+        group_id : str
+            The ID of the group to which this message is associated. In some settings,
+            this is used as the FL round.
+        ttl : str
+            Time-to-live for the round trip of this message, i.e., the time from sending
+            this message to receiving a reply. It specifies the duration for which the
+            message and its potential reply are considered valid.
+
+        Returns
+        -------
+        Message
+            A new `Message` instance with the specified content and metadata.
+        """
+        _, run_id = self._get_grpc_driver_and_run_id()
+        metadata = Metadata(
+            run_id=run_id,
+            message_id="",
+            src_node_id=self.node.node_id,
+            dst_node_id=dst_node_id,
+            reply_to_message="",
+            group_id=group_id,
+            ttl=ttl,
+            message_type=message_type,
+        )
+        return Message(metadata=metadata, content=content)
+
+    def get_node_ids(self) -> List[int]:
         """Get node IDs."""
         grpc_driver, run_id = self._get_grpc_driver_and_run_id()
-
         # Call GrpcDriver method
         res = grpc_driver.get_nodes(GetNodesRequest(run_id=run_id))
-        return list(res.nodes)
+        return [node.node_id for node in res.nodes]
 
-    def push_task_ins(self, task_ins_list: List[TaskIns]) -> List[str]:
-        """Schedule tasks."""
-        grpc_driver, run_id = self._get_grpc_driver_and_run_id()
+    def push_messages(self, messages: Iterable[Message]) -> Iterable[str]:
+        """Push messages to specified node IDs.
+
+        This method takes an iterable of messages and sends each message
+        to the node it specifies in `node_id`.
 
-        # Set run_id
-        for task_ins in task_ins_list:
-            task_ins.run_id = run_id
+        Parameters
+        ----------
+        messages : Iterable[Message]
+            An iterable of messages to be sent.
 
+        Returns
+        -------
+        message_ids : Iterable[str]
+            An iterable of IDs for the messages that were sent, which can be used
+            to pull responses.
```

Comment:
How do we indicate errors? For example, if two messages are sent via `push_messages`, but one of them has an invalid `dst_node_id` specified, what does the response look like?

Reply:
For now, if one of the message has incorrect `Metadata` (which usually means the code is wrong), an exception will be raised. Should we do something instead of raising an exception to keep the ServerApp running?

Reply:
True, what you mention is our client-side safety net. The SuperLink also has the capability to reject messages, in which case it returns an empty `str` as the message ID. That's compatible with the current design, so I'm resolving this comment.